x/qgb/orchestrator/orchestrator_client.go

```diff
@@ -194,43 +193,58 @@ func (oc *orchestratorClient) SubscribeDataCommitment(ctx context.Context) (<-ch
 			select {
 			case <-ctx.Done():
 				return
-			case ev := <-results:
-				eventDataHeader := ev.Data.(coretypes.EventDataNewBlockHeader)
-				height := uint64(eventDataHeader.Header.Height)
-				// todo: refactor to ensure that no ranges of blocks are missed if the
-				// parameters are changed
-				if height%types.DataCommitmentWindow != 0 {
+			default:
+				latestDCNonce, err := oc.querier.QueryLatestDataCommitmentNonce(ctx)
+				if err != nil {
+					oc.logger.Error(err.Error())
+					time.Sleep(5 * time.Second)
 					continue
 				}
 
-				// TODO: calculate start height some other way that can handle changes
-				// in the data window param
-				startHeight := height - types.DataCommitmentWindow
-				endHeight := height
+				// query data commitment request
+				dc, err := oc.querier.QueryDataCommitmentByNonce(ctx, latestDCNonce)
+				if err != nil {
+					oc.logger.Error(err.Error())
+					time.Sleep(5 * time.Second)
+					continue
+				}
+				if dc == nil {
+					time.Sleep(5 * time.Second)
+					continue
+				}
```

Comment:
will this just loop and query every 5 seconds?

I'm not the biggest fan of hardcoding these timeouts, but I'm okay with keeping them since we will certainly change them with the worker refactor

Reply:
yes, will change, that's why I am adding it  for now.